x/icqueries/types/query.go

```diff
@@ -1,38 +1,16 @@
 package types
 
 import (
-	"strings"
-
-	errorsmod "cosmossdk.io/errors"
+	"fmt"
+	"time"
 )
 
-func (query *Query) ValidateBasic() error {
-	if len(query.GetId()) <= 0 {
-		return errorsmod.Wrapf(ErrInvalidQuery, "query id is required")
-	}
-
-	if len(query.GetConnectionId()) <= 0 {
-		return errorsmod.Wrapf(ErrInvalidQuery, "connection id is required")
-	}
-
-	if !strings.HasPrefix(query.GetConnectionId(), "connection") {
-		return errorsmod.Wrapf(ErrInvalidQuery, "connection id must start with connection")
-	}
-
-	if len(query.GetChainId()) <= 0 {
-		return errorsmod.Wrapf(ErrInvalidQuery, "chain id is required")
-	}
-
-	if len(query.GetQueryType()) <= 0 {
-		return errorsmod.Wrapf(ErrInvalidQuery, "query type is required")
-	}
-
-	if len(query.GetRequest()) <= 0 {
-		return errorsmod.Wrapf(ErrInvalidQuery, "query request is required")
-	}
+// HasTimedOut Check if a query has timed-out by checking whether the block time is after  the timeout timestamp
+func (q *Query) HasTimedOut(currentBlockTime time.Time) bool {
+	return q.TimeoutTimestamp < uint64(currentBlockTime.UnixNano())
+}
 
-	if len(query.GetCallbackId()) <= 0 {
-		return errorsmod.Wrapf(ErrInvalidQuery, "callback id is required")
-	}
-	return nil
+// Description Prints an abbreviated query description for logging purposes
+func (q *Query) Description() string {
```

Comment:
Maybe I missed it, but I couldn't find anywhere where this function is used? Do we need it?